packages/components/src/custom-select-control/test/index.js

```diff
@@ -262,6 +271,81 @@ describe.each( [
 		expect( screen.getByRole( 'option', { name: /hint/i } ) ).toBeVisible();
 	} );
 
+	it( 'Should return object onChange', async () => {
+		const user = userEvent.setup();
+		const mockOnChange = jest.fn();
+
+		render(
+			<Component
+				{ ...props }
+				value={ props.options[ 0 ] }
+				onChange={ mockOnChange }
+			/>
+		);
+
+		await user.click(
+			screen.getByRole( 'button', {
+				expanded: false,
+			} )
+		);
+
+		// DIFFERENCE WITH V2: NOT CALLED
```

Comment:
If this is something that we need to update, let's add a TODO comment to remember to address it

Reply:
I'm not sure about how much we can do about this, since:
- we don't want to change the v1 behaviour (which doesn't call `onChange` on initial render)
- the v2 behaviour comes directly from `Ariakit` — I'm not sure how easy it would be to suppress _only_ the first call without introducing unwanted side effects.

I will add an item to the [tracking issue](https://github.com/WordPress/gutenberg/issues/55023)

Reply:
Good news! This will be fixed in https://github.com/WordPress/gutenberg/pull/62733